### PR TITLE
kubeadm: add note about a bug in the PublicKeysECDSA FG

### PR DESCRIPTION
**Pull request by neolit123**
The PublicKeysECDSA has been poorly tested and supported and apparently it had a bug where keys in kubeconfig files were using RSA even if the FG was true. (even the FG name is misleading...)

Add note about that in the FG section of the kubeadm init doc.

xref https://github.com/kubernetes/kubeadm/issues/3072

### CODE REVIEW

**Comment by neolit123** [comment]
i don't think this should be in the dev-1.31 branch as it's not exactly adding new behavior?

/sig cluster-lifecycle
/kind bug
/cc @pacoxu @carlory 
/assign @sftim

**Review by pacoxu — APPROVED** [review]
/lgtm

This is a known issue now.

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (sftim) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:179]: You can merge this into _main_ (and also request backports)
```markdown
switch between the RSA and ECDSA algorithms on the fly or during upgrades.
Kubernetes {{< skew currentVersion >}} has a bug where keys in generated kubeconfig
files are set use RSA despite the feature gate being enabled.
```
(or some similar text)


and then once that merges into _dev-1.31_, propose a new PR to use the text here:
```markdown
switch between the RSA and ECDSA algorithms on the fly or during upgrades. Kubernetes versions before
v1.31 had a bug where keys in generated kubeconfig files were set use RSA, even when you had
enabled the `PublicKeysECDSA` feature gate.
```

Bit more complicated to do, but it's more complete and helps people who are using a release other than the very latest.

---
By “request backports” I mean file an issue asking for docs contributors to backport the docs change to older release docs.

**Comment by sftim** [comment]
/hold

We [avoid statements about the future](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future) and that includes most promises about future releases.

If we make an exception here, let's be clear about why.

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (pacoxu) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:179]: Sound good to me.

Comment (neolit123) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]: > You can merge this into main (and also request backports)

updated to include this text

**Review by mengjiao-liu — COMMENTED** [review]
[The preview page](https://deploy-preview-46795--kubernetes-io-main-staging.netlify.app/docs/reference/setup-tools/kubeadm/kubeadm-init/) looks good to me.
/lgtm

**Comment by pacoxu** [comment]
/lgtm

image

**Pull request merged**
